src/uniprotkb/components/protein-data-views/SubCellViz.tsx

```diff
@@ -62,6 +66,17 @@ const getGoTermClassNames = (locationGroup: Element) =>
     .filter((className) => className.startsWith('GO'))
     .map((goId) => `.${goId}`);
 
+const getUniProtTextSelectors = (subcellularPresentSVG: Element) =>
+  [
+    `#${subcellularPresentSVG.id}term`,
+    ...Array.from(subcellularPresentSVG.classList)
+      .map((className: string) => {
+        const id = className.match(reMpPart)?.groups?.id;
+        return id && `#${id}term`;
+      })
+      .filter(Boolean),
+  ] as string[];
```

Comment:
Needed `as string[]` otherwise it thinks there could be come `undefined` elements which I don't think is possible with `filter(Boolean)`. Open to improvements.

Reply:
I would suggest going for a function returning a type predicate instead of just the Boolean constructor
```ts
const getUniProtTextSelectors = (subcellularPresentSVG: Element): string[] => [
  `#${subcellularPresentSVG.id}term`,
  ...Array.from(subcellularPresentSVG.classList)
    .map((className: string) => {
      const id = className.match(reMpPart)?.groups?.id;
      return id && `#${id}term`;
    })
    .filter((sel: string | undefined): sel is string => Boolean(sel)),
];
See https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates for details
```